src/algebra/zset/mod.rs

```diff
@@ -82,6 +82,21 @@ where
 /// An implementation of ZSets using [`FiniteHashMap`]s
 pub type ZSetHashMap<Data, Weight> = FiniteHashMap<Data, Weight>;
 
+/// Build a Z-set from an iterator over values by adding each value with
+/// weight 1.
+impl<Data, Weight> FromIterator<Data> for ZSetHashMap<Data, Weight>
+where
+    Data: KeyProperties,
+    Weight: ZRingValue,
+{
+    fn from_iter<T>(iter: T) -> Self
+    where
+        T: IntoIterator<Item = Data>,
+    {
+        Self::from_iter(iter.into_iter().map(|d| (d, Weight::one())))
```

Comment:
Is this infinitely recursive? It looks like it calls itself again

Reply:
It calls a different `FromIterator` implementation, which takes an iterator over (value, weight) pairs.